my/smscalls.py

```diff
@@ -49,3 +47,48 @@ def calls() -> Iterator[Call]:
                 continue
             emitted.add(c.dt)
             yield c
+
+
+class Message(NamedTuple):
+    dt: datetime
+    who: str
+    message: str
+    phone_number: str
+    from_me: bool
+
+
+def messages() -> Iterator[Message]:
+    files = get_files(config.export_path, glob='sms-*.xml')
+
+    emitted: Set[Tuple[datetime, str, bool]] = set()
+    for p in files:
+        for c in _extract_messages(p):
+            key = (c.dt, c.who, c.from_me)
+            if key in emitted:
+                continue
+            emitted.add(key)
+            yield c
+
+
+def _extract_messages(path: Path) -> Iterator[Message]:
+    tr = etree.parse(str(path))
+    for mxml in tr.findall('sms'):
+        yield Message(
+            dt=_parse_dt_ms(mxml.get('date')),
+            who=mxml.get('contact_name'),
+            message=mxml.get('body'),
+            phone_number=mxml.get('address'),
+            from_me=mxml.get('type') == '2',  # 1 is received message, 2 is sent message
+        )
+
+def _parse_dt_ms(d: str) -> datetime:
+    return pytz.utc.localize(datetime.utcfromtimestamp(int(d) / 1000))
```

Comment:
Wonder why I wasn't using `datetime.fromtimestamp(ts, tz=pytz.utc)` in the first place across my code.. should be equivalent, but a bit more straightforward.

Reply:
I believe `datetime.fromtimestamp` converts it to your own timezone, while `datetime.utcfromtimestamp` converts it to a datetime unaware utc timestamp.

If thats the case and you provide UTC, it should still use UTC.

At that point, it makes more sense to just do

```
>>> from datetime import datetime, timezone
>>> datetime.fromtimestamp(1601577294, tz=timezone.utc)
datetime.datetime(2020, 10, 1, 18, 34, 54, tzinfo=datetime.timezone.utc)
```

removing `pytz` altogether, and just using the builtin `datetime.timezone` objects.

Im unsure on why one would use `pytz` as well. If you don't have a reason here I can swap it to the stdlib version.

Reply:
Huh, interesting, somehow wasn't aware of `timezone.utc`! Don't mind replacing, thanks.